src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -560,6 +568,7 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
             Prompt = GetDefaultPrompt(prompt);
             MaxLength = len;
 
+            IsProcessing = contexts.Count != 1;
```

Comment:
This is the check that InterruptR command has been performing. I am open to suggestions. Process can start and may continue as long as user input does not require more package installs. I.e. x <- 1 is perfectly safe. If it does, the session will be canceled before package installation begins.

Not sure about evals, I guess it may be possible to do something like `eval("install.packages").